mlflow/types/schema.py

```diff
@@ -408,3 +408,158 @@ def __eq__(self, other) -> bool:
 
     def __repr__(self) -> str:
         return repr(self.inputs)
+
+
+class ParamSpec:
+    """
+    Specification used to represent parameters for the model.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        type: str,  # pylint: disable=redefined-builtin
+        default: Optional[Any] = None,
+        optional: bool = True,
```

Comment:
@serena-ruan Based on discussion from the design doc and feedback from @mehtayogita , can we remove the `optional` field for now and make `default` a required argument? 

Initially, I think it's simpler if every param is optional and defines a default. We can always add `optional = false` support later :)